plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1914,12 +1748,223 @@ bool producer_plugin_impl::remove_expired_blacklisted_trxs( const fc::time_point
+                 ("entire_trace", get_trace(trace, except_ptr)));
+      } else {
+         fc_dlog(_trx_log, "[TRX_TRACE] Speculative execution is ACCEPTING tx: ${trx}",
+                 ("trx", chain_plug->get_log_trx(trx->get_transaction())));
+         fc_dlog(_trx_successful_trace_log, "[TRX_TRACE] Speculative execution is ACCEPTING tx: ${txid}, auth: ${a}, cpu: ${cpu}",
+                 ("txid", trx->id())("a", trx->get_transaction().first_authorizer())
+                 ("cpu", billed_cpu_us));
+         fc_dlog(_trx_trace_success_log, "[TRX_TRACE] Speculative execution is ACCEPTING tx: ${entire_trace}",
+                 ("entire_trace", get_trace(trace, except_ptr)));
+      }
+   }
+}
+
+// Does not modify unapplied_transaction_queue
+producer_plugin_impl::push_result
+producer_plugin_impl::push_transaction( const fc::time_point& block_deadline,
+                                        const transaction_metadata_ptr& trx,
+                                        bool persist_until_expired,
+                                        bool return_failure_trace,
+                                        next_function<transaction_trace_ptr> next )
+{
+   auto start = fc::time_point::now();
+   fc_dlog( _trx_successful_trace_log, "Time since last trx: ${t}us", ("t", start - _idle_trx_time) );
+
+   auto first_auth = trx->packed_trx()->get_transaction().first_authorizer();
+   if( _pending_block_mode == pending_block_mode::producing && _account_fails.failure_limit( first_auth ) ) {
+      if( next ) {
+         auto except_ptr = std::static_pointer_cast<fc::exception>( std::make_shared<tx_cpu_usage_exceeded>(
+               FC_LOG_MESSAGE( error, "transaction ${id} exceeded failure limit for account ${a}",
+                               ("id", trx->id())( "a", first_auth ) ) ) );
+         log_trx_results( trx, nullptr, except_ptr, start );
+         next( except_ptr );
+      }
+      _idle_trx_time = fc::time_point::now();
+      return push_result{.failed = true};
+   }
+
+   chain::controller& chain = chain_plug->chain();
+   fc::microseconds max_trx_time = fc::milliseconds( _max_transaction_time_ms.load() );
+   if( max_trx_time.count() < 0 ) max_trx_time = fc::microseconds::maximum();
+
+   bool disable_subjective_billing = ( _pending_block_mode == pending_block_mode::producing )
+                                     || ( persist_until_expired && _disable_subjective_api_billing )
+                                     || ( !persist_until_expired && _disable_subjective_p2p_billing )
+                                     || trx->read_only;
+
+   uint32_t sub_bill = 0;
+   if( !disable_subjective_billing )
+      sub_bill = _subjective_billing.get_subjective_bill( first_auth, fc::time_point::now() );
+
+   auto prev_billed_cpu_time_us = trx->billed_cpu_time_us;
+   if( prev_billed_cpu_time_us > 0 && _pending_block_mode == pending_block_mode::producing ) {
```

Comment:
move `_pending_block_mode == pending_block_mode::producing` as first check of `if`